### PR TITLE
Add org.nicotine_plus.Nicotine (Nicotine+)

### PR DESCRIPTION
**Pull request by mathiascode**
Shouldn't be merged until we release Nicotine+ 2.0.0 in the near future, but I'm creating this PR early to check if any changes are needed upstream.

Closes https://github.com/Nicotine-Plus/nicotine-plus/issues/102

### CODE REVIEW

File: org.nicotine_plus.Nicotine.json [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        "/bin"
+      ]
+    },
+    {
+      "name" : "nicotine",
+      "buildsystem" : "simple",
+      "sources" : [
+        {
+          "url" : "https://github.com/Nicotine-Plus/nicotine-plus.git",
+          "commit" : "2e4f3b327d9f68a643d4af1c50570695cd07c1a3",
```
Comment (mathiascode) [inline org.nicotine_plus.Nicotine.json:72]: Will change this to a release tag later.

File: org.nicotine_plus.Nicotine.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    "/lib/pkgconfig",
+    "/share/aclocal",
+    "/share/info",
+    "/share/man",
+    "/share/gtk-doc",
+    "/share/vala",
+    "*.a",
+    "*.la"
+  ],
+  "finish-args" : [
```
Comment (bilelmoussaoui) [inline org.nicotine_plus.Nicotine.json:93]: We tend to have the finish-args & then cleanup before the modules. It makes review/contributions easier

**Comment by bilelmoussaoui** [comment]
bot, build org.nicotine_plus.Nicotine

**Comment by mathiascode** [comment]
bot, build org.nicotine_plus.Nicotine

**Comment by mathiascode** [comment]
https://github.com/Nicotine-Plus/nicotine-plus/pull/291

**Comment by mathiascode** [comment]
bot, build org.nicotine_plus.Nicotine

**Comment by LeandroStanger** [comment]
bot, build org.nicotine_plus.Nicotine

**Comment by mathiascode** [comment]
What would be the best way to get the correct system prefix in Python (/app instead of /usr)? Plugins, tray icons and sound effects are stored in /app/share/nicotine/, but the code currently uses the sys.prefix call, which returns /usr.

**Comment by bilelmoussaoui** [comment]
> What would be the best way to get the correct system prefix in Python (/app instead of /usr)? Plugins, tray icons and sound effects are stored in /app/share/nicotine/, but the code currently uses the sys.prefix call, which returns /usr.
> 
> Edit: Here's what I came up with so far, but it feels a bit dirty: [Nicotine-Plus/nicotine-plus#296](https://github.com/Nicotine-Plus/nicotine-plus/pull/296)

The right way is to store the prefix you have passed to setuptools when installing the application in a pre-configured file. The current solution looks okay to me though

**Comment by mathiascode** [comment]
bot, build org.nicotine_plus.Nicotine

**Comment by nedrichards** [comment]
bot, build org.nicotine_plus.Nicotine

**Comment by mathiascode** [comment]
bot, build org.nicotine_plus.Nicotine

**Comment by mathiascode** [comment]
bot, build org.nicotine_plus.Nicotine

**Comment by mathiascode** [comment]
Alright, this should be ready now. :)

**Comment by bilelmoussaoui** [comment]
Anyone else from upstream that would like to have write access to the repository once created?

**Comment by mathiascode** [comment]
@kiplingw Are you interested?

**Comment by kiplingw** [comment]
Thanks @mathiascode. I'll merge this, although I am generally opposed to Flatpak for philosophical reasons. I don't think it holds a candle stick to Apt, nor do I think shipping runtimes dependencies within a package is efficient or sensible either. Nevertheless, I understand not everyone uses Debian based distributions.

**Comment by mathiascode** [comment]
@bilelmoussaoui You can give me write access in the new repo. If someone else from the team, or future contributors, want write access in the future, we can open an issue.

**Pull request closed without merging**